_studio/shared/src/libmfx_allocator_vaapi.cpp

```diff
@@ -279,196 +279,171 @@ mfxDefaultAllocatorVAAPI::AllocFramesHW(
+
+        response->mids           = pSelf->m_frameHandles.data();
+        response->NumFrameActual = request->NumFrameSuggested;
+        response->AllocId        = request->AllocId;
+
+        return MFX_ERR_NONE;
     }
 
-    // allocate frames in cycle
-    surfaces = (VASurfaceID*)calloc(surfaces_num, sizeof(VASurfaceID));
-    vaapi_mids = (vaapiMemIdInt*)calloc(surfaces_num, sizeof(vaapiMemIdInt));
-    if ((NULL == surfaces) || (NULL == vaapi_mids))
+    // Use temporary storage for preliminary operations. If some of them fail, current state of allocator remain unchanged.
+    // When allocation will be finished, just move content of these vectors to internal allocator storage
+    std::vector<VASurfaceID>   AllocatedSurfaces(request->NumFrameSuggested, VA_INVALID_ID);
+    std::vector<vaapiMemIdInt> AllocatedMids(request->NumFrameSuggested);
+
+    VAStatus  va_res  = VA_STATUS_SUCCESS;
+    mfxStatus mfx_res = MFX_ERR_NONE;
+
+    if( VA_FOURCC_P208 != va_fourcc)
     {
-        mfx_res = MFX_ERR_MEMORY_ALLOC;
-    }
+        unsigned int format;
+        std::vector<VASurfaceAttrib> attrib;
+        FillSurfaceAttrs(attrib, format, request->Info.FourCC, va_fourcc, request->Type);
 
-    if (MFX_ERR_NONE == mfx_res)
+        va_res = vaCreateSurfaces(pSelf->pVADisplay,
+                            format,
+                            request->Info.Width, request->Info.Height,
+                            AllocatedSurfaces.data(),
+                            AllocatedSurfaces.size(),
+                            attrib.data(),
+                            attrib.size());
+
+        MFX_CHECK(va_res == VA_STATUS_SUCCESS, VA_TO_MFX_STATUS(va_res));
+    }
+    else
     {
-        if( VA_FOURCC_P208 != va_fourcc)
-        {
-            unsigned int format;
-            std::vector<VASurfaceAttrib> attrib;
-            FillSurfaceAttrs(attrib, format,  fourcc, va_fourcc, request->Type);
-
-            va_res = vaCreateSurfaces(pSelf->pVADisplay,
-                                format,
-                                request->Info.Width, request->Info.Height,
-                                surfaces,
-                                surfaces_num,
-                                attrib.data(),
-                                attrib.size());
+        mfxU32 codedbuf_size, codedbuf_num;
+        VABufferType codedbuf_type;
 
-            mfx_res = VA_TO_MFX_STATUS(va_res);
-            bCreateSrfSucceeded = (MFX_ERR_NONE == mfx_res);
+        if (request->Info.FourCC == MFX_FOURCC_VP8_SEGMAP)
+        {
+            codedbuf_size = request->Info.Width;
+            codedbuf_num  = request->Info.Height;
+            codedbuf_type = VAEncMacroblockMapBufferType;
         }
         else
         {
-            VAContextID context_id = request->AllocId;
-            mfxU32 codedbuf_size, codedbuf_num;
-            VABufferType codedbuf_type;
+            int width32  = 32 * ((request->Info.Width  + 31) >> 5);
```

Comment:
`width32/height32` doesn't look to be good names. It would be better to use something like `aligned_width` or just add a comment that we need aligned value. BTW, we have a lot of alignment functions/macros like `UMC::align_value`, using of them would make code more readable than `32 * ((request->Info.Width  + 31) >> 5)`.

Reply:
This is for separate PR. I know several places in code where we have exactly same calculations. I suppose we need to change them all in separate PR

Reply:
addressed in #1265